support Ubuntu 24.04 LTS

libglm-dev version there is 0.9.9.8+ds-7

during compilation we got following complain

```
$ scons test_expand
scons: Reading SConscript files ...
scons: done reading SConscript files.
scons: Building targets ...
g++ -o test_expand.o -c -std=c++17 -g -O0 -Wall -Wno-sign-compare -DGLM_FORCE_RADIANS -DGLM_ENABLE_EXPERIMENTAL -I. test_expand.cpp
In file included from box2.hpp:4,
                 from test_expand.cpp:4:
boost_geometry_adapt.hpp:12:18: fatal error: glm/detail/type_vec.hpp: No such file or directory
   12 |         #include <glm/detail/type_vec.hpp>
      |                  ^~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
scons: *** [test_expand.o] Error 1
scons: building terminated because of errors.
```

--- a/geometry/boost_geometry_adapt.hpp
+++ b/geometry/boost_geometry_adapt.hpp
@@ -1,4 +1,7 @@
-/*! Adaptuje štruktúry s knižnice glm (0.9.6.3, 0.9.9.0) pre použitie s boost::geometry. */
+/*! \file
+Adapts glm vector types (vec1, vec2 and vec3) for boost::geometry.
+
+Supported glm versions are 0.9.9.8 (Ubuntu 24.04 LTS), 0.9.9.0 and 0.9.6.3. */
 #pragma once
 #include <boost/mpl/int.hpp>
 #include <boost/geometry/core/access.hpp>
@@ -8,16 +11,23 @@
 #include <boost/geometry/core/coordinate_dimension.hpp>
 #include <glm/detail/setup.hpp>
 
-#if GLM_VERSION >= 990
+// for Ubuntu 24.04 LTS
+#if GLM_VERSION >= 998
+	#include <glm/detail/type_vec1.hpp>
+	#include <glm/detail/type_vec2.hpp>
+	#include <glm/detail/type_vec3.hpp>
+
+// for Ubuntu 18.04 LTS
+#elif GLM_VERSION >= 990
 	#include <glm/detail/type_vec.hpp>
-#else
+#else  // for glm 0.9.6.3
 	#include <glm/vec2.hpp>
 	#include <glm/vec3.hpp>
 #endif
 
-#if GLM_VERSION >= 990
+#if GLM_VERSION >= 998
 
-namespace boost { namespace geometry { namespace traits {
+namespace boost::geometry::traits {
 
 // glm::vec adaptation
 template <glm::length_t L, typename T, glm::qualifier Q>
@@ -52,7 +62,7 @@ struct access<glm::vec<L, T, Q>, Dimension>
 	static void set(point_type & p, value_type const & v) {p[Dimension] = v;}
 };
 
-}}}  // geometry, boost, traits
+}  // boost::geometry::traits
 
 #else  // glm 0.9.6.3
 

--- a/geometry/readme.md
+++ b/geometry/readme.md
@@ -1,6 +1,15 @@
 # geometry
 Implmentuje geometricke algoritmy štruktúry a pomocné funkcie.
 
+> TODO: What we have there? How to use stafff we have there?
+
+- `utility.hpp`
+- `box2.hpp`
+- `box3.hpp`
+- `glmprint.hpp`
+- `boost_geometry_adapt.hpp`
+
+> TODO: I rememer that I've renamed this to `glm_adapt.hpp`, but
 
 ## geometry:inštalácia
 

